### ISSUE TITLE
Eval bug: inference stops when interaction become long

### ISSUE THREAD

**Issue by apalazzi** [ggml-org/llama.cpp#19636]
### Name and Version

$ llama-cli --version
load_backend: loaded RPC backend from /usr/local/bin/libggml-rpc.so
ggml_vulkan: Found 2 Vulkan devices:
ggml_vulkan: 0 = AMD Radeon (TM) Pro VII (RADV VEGA20) (radv) | uma: 0 | fp16: 1 | bf16: 0 | warp size: 64 | shared memory: 65536 | int dot: 1 | matrix cores: none
ggml_vulkan: 1 = AMD Radeon Graphics (RADV VEGA20) (radv) | uma: 0 | fp16: 1 | bf16: 0 | warp size: 64 | shared memory: 65536 | int dot: 1 | matrix cores: none
load_backend: loaded Vulkan backend from /usr/local/bin/libggml-vulkan.so
load_backend: loaded CPU backend from /usr/local/bin/libggml-cpu-haswell.so
version: 7973 (39bf692af)
built with GNU 11.4.0 for Linux x86_64

### Operating systems

Linux

### GGML backends

Vulkan

### Hardware

Ryzen 5800x + MI50 32GB + MI50 16GB

### Models

lmstudio-community_DeepSeek-R1-Distill-Qwen-32B-GGUF_DeepSeek-R1-Distill-Qwen-32B-Q4_K_M.gguf

### Problem description & steps to reproduce

I use llama-server launched by llama-swap and with open-webui; when the chat gets long llama.cpp stops with the error "request (16203 tokens) exceeds the available context size (13312 tokens), try increasing it", even though context seems big enough (see log).

### Relevant log output

<details>
<summary>Logs</summary>
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: constructing llama_context                                                                                                                                           │
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: n_ctx is not divisible by n_seq_max - rounding down to 133120                                                                                                        │
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: n_seq_max     = 10                                                                                                                                                   │
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: n_ctx         = 133120                                                                                                                                               │
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: n_ctx_seq     = 13312                                                                                                                                                │
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: n_batch       = 2048                                                                                                                                                 │
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: n_ubatch      = 512                                                                                                                                                  │
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: causal_attn   = 1                                                                                                                                                    │
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: flash_attn    = enabled                                                                                                                                              │
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: kv_unified    = false                                                                                                                                                │
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: freq_base     = 1000000.0                                                                                                                                            │
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: freq_scale    = 1                                                                                                                                                    │
Feb 14 21:09:07 delphi llama-swap[779120]: llama_context: n_ctx_seq (13312) < n_ctx_train (131072) -- the full capacity of the model will not be utilized
[...]
Feb 15 07:52:25 delphi llama-swap[779120]: slot      release: id  0 | task 16432 | stop processing: n_tokens = 3266, truncated = 0                                                                                             │udev on /dev type devtmpfs (rw,nosuid,relatime,size=16391408k,nr_inodes=4097852,mode=755,inode64)
Feb 15 07:52:25 delphi llama-swap[779120]: srv  update_slots: all slots are idle                                                                                                                                               │devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
Feb 15 07:52:25 delphi llama-swap[779120]: srv  log_server_r: done request: POST /v1/chat/completions 127.0.0.1 200                                                                                                            │tmpfs on /run type tmpfs (rw,nosuid,nodev,noexec,relatime,size=3286476k,mode=755,inode64)
Feb 15 07:52:25 delphi llama-swap[779120]: srv  params_from_: Chat format: DeepSeek R1                                                                                                                                         │bpf on /sys/fs/bpf type bpf (rw,nosuid,nodev,noexec,relatime,mode=700)
Feb 15 07:52:25 delphi llama-swap[779120]: slot get_availabl: id  3 | task -1 | selected slot by LCP similarity, sim_best = 0.307 (> 0.100 thold), f_keep = 0.502                                                              │tmpfs on /run/user/1000 type tmpfs (rw,nosuid,nodev,relatime,size=3286472k,nr_inodes=821618,mode=700,uid=1000,gid=1000,inoFeb 15 07:52:25 delphi llama-swap[779120]: slot launch_slot_: id  3 | task -1 | sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist                 │de64)
Feb 15 07:52:25 delphi llama-swap[779120]: slot launch_slot_: id  3 | task 16787 | processing task, is_child = 0                                                                                                               │andrea@delphi:~$ mount|grep model
Feb 15 07:52:25 delphi llama-swap[779120]: slot update_slots: id  3 | task 16787 | new prompt, n_ctx_slot = 13312, n_keep = 0, task.n_tokens = 16203                                                                           │andrea@delphi:~$ sudo nano /etc/fstab
Feb 15 07:52:25 delphi llama-swap[779120]: srv    send_error: task id = 16787, error: request (16203 tokens) exceeds the available context size (13312 tokens), try increasing it                                              │[sudo] password for andrea:
Feb 15 07:52:25 delphi llama-swap[779120]: srv          stop: cancel task, id_task = 16787                                                                                                                                     │andrea@delphi:~$
Feb 15 07:52:25 delphi llama-swap[779120]: slot      release: id  3 | task 16787 | stop processing: n_tokens = 9920, truncated = 0                                                                                             │andrea@delphi:~$
Feb 15 07:52:25 delphi llama-swap[779120]: srv  update_slots: no tokens to decode                                                                                                                                              │andrea@delphi:~$ llama-cli --version
Feb 15 07:52:25 delphi llama-swap[779120]: srv  update_slots: all slots are idle
</details>

**Comment by amakropoulos** [comment]
hi, this seems related to #17284

**Comment by apalazzi** [comment]
In fact, it seems to be the same issue. Thanks.
